### PR TITLE
First draft of frontend histogram component

### LINKED ISSUE #28: Add a cases histogram to the frontend
The frontend needs some sort of React component which can display an interactive histogram of cases.

Some possible options for components [are here](https://dev.to/giteden/top-5-react-chart-libraries-for-2020-1amb)

## Prerequisites
- [ ] depends on #32

### PR DESCRIPTION
**Pull request by crangelsmith**
Closes #28 

*Update*:

After talking to @jemrobinson  a couple of to-dos before merging this Pr:

- [x] Check if recharts can handle two datasets
- [x] Fixing main grid shape to percentage instead of absolute values.

### CODE REVIEW

**Comment by crangelsmith** [comment]
This is how the frontend is currently looking (histogram has dummy data):

Landing page:

Screenshot 2021-01-27 at 10 20 10

After clicking on a country:
Screenshot 2021-01-27 at 10 20 27

There is a lot of asthetic/desing improvements that should be done, but some questions for @KFArnold and @jemrobinson:

-  Any opinion on the colours we are using?
-  What else do we want to show in the histogram component?
-  We need to show a title somewhere that describes what is this dashboard about. 

Before moving further in the front end maybe we need to define the design that we want in each of the components.

File: frontend/src/components/Histogram.jsx [hunk 1]
```diff
@@ -1,7 +1,232 @@
...
+  CartesianGrid,
+  Tooltip,
+  Legend,
+  ComposedChart,
+  Bar,
+} from "recharts";
+
+const data = [
+  {
+    name: "2020-03-01",
```
Comment (jemrobinson) [inline frontend/src/components/Histogram.jsx]: Can you rename these to eg. `date`, `cases_real` and `cases_counterfactual`?

File: frontend/src/components/MainGrid.jsx [hunk 2]
```diff
@@ -30,9 +35,18 @@ export default class MainGrid extends React.Component {
   }
 
   // Update the state for a new country
-  handleCountryChange(iso_code) {
+  handleCountryChange(iso_code, selectedCaseNumber) {
```
Comment (jemrobinson) [inline frontend/src/components/MainGrid.jsx]: I wonder whether something like `currentTotalCases` would be clearer than `selectedCaseNumber`?

File: frontend/src/components/WorldMap.jsx [hunk 3]
```diff
@@ -16,12 +16,15 @@ const WorldMap = ({ countries, onCountrySelect }) => {
     const confirmedText = country.properties.confirmedText;
     layer.bindPopup(`${name} ${confirmedText}`);
     layer.on("click", function (e) {
-      onCountrySelect(e.target.feature.properties.iso_code);
+      onCountrySelect(
+        e.target.feature.properties.name,
+        e.target.feature.properties.confirmedText
```
Comment (jemrobinson) [inline frontend/src/components/WorldMap.jsx]: I wonder whether we should rename `confirmed` and `confirmedText` to eg. `totalCasesPerMillion` and `totalCasesPerMillionText`?

**Review by jemrobinson — APPROVED** [review]
Merge whenever you're ready - I just have a couple of minor questions about variable names.

File: frontend/src/components/Histogram.jsx (hunk 1 above)
Comment (crangelsmith) [reply frontend/src/components/Histogram.jsx]: Done!

File: frontend/src/components/MainGrid.jsx (hunk 2 above)
Comment (crangelsmith) [reply frontend/src/components/MainGrid.jsx]: Done!

File: frontend/src/components/WorldMap.jsx (hunk 3 above)
Comment (crangelsmith) [reply frontend/src/components/WorldMap.jsx]: And done :)

**Pull request merged**